x/gamm/keeper/multihop_test.go

```diff
@@ -193,6 +283,13 @@ func (suite *KeeperTestSuite) TestBalancerPoolSimpleMultihopSwapExactAmountOut()
 				sp := tokenInAmount.ToDec().Quo(test.param.tokenOut.Amount.ToDec())
 				fmt.Printf("spBefore %s, spAfter %s, sp actual %s\n", spotPriceBefore, spotPriceAfter, sp)
 				suite.True(sp.GT(spotPriceBefore) && sp.LT(spotPriceAfter), "multi-hop spot price wrong, test: %v", test.name)
+
+				if test.reducedFeeApplied {
+					suite.Require().True(tokenInAmount.LT(tokenInCalculatedAsSeparateSwaps.Amount))
+					suite.Require().True(tokenInAmount.GT(tokenInCalculatedAsSeparateSwapsWithoutFee.Amount))
```

Comment:
Does it matter if `reducedFee` is applied or not if the swap fee is 0?

Reply:
Cause i think `suite.Require().True(tokenInAmount.GT(tokenInCalculatedAsSeparateSwapsWithoutFee.Amount))` doesnot have to be in the if statement

Reply:
if swap fee is 0, than fee will not reduced. 
this line is just one more defensive check, that we will not lost user assets due some miscode with fee reduction (ex: after fee reduction another user could get more assets, that raw swap without fee at all)

Reply:
we can change it to `GTE`

Reply:
> (ex: after fee reduction another user could get more assets, that raw swap without fee at all)

How is this possible? If we have fee reduction, there is still a `swapFee` so, how can a user get more assets than `tokenInAmount`?

Reply:
> > (ex: after fee reduction another user could get more assets, that raw swap without fee at all)
> 
> How is this possible? If we have fee reduction, there is still a `swapFee` so, how can a user get more assets than `tokenInAmount`?

just remember a june case, when users was able to get 50% from pools :) By design it is not possible, but one math code mistake, that was not covered my simple unit test make it real.

Reply:
aah makes sense! You got me there 😁